src/main/java/com/zzpj/dtos/BookDto.java

```diff
@@ -11,6 +11,8 @@
 @JsonPropertyOrder({"name", "author"})
 public class BookDto {
 
+    public Long id;
```

Comment:
nie wczytywałam się mocno, ale pewnie trzeba będzie zrobić 2 osobne DTO - z id i bez, bo przy aktualizacji faktycznie jest, ale przy dodawaniu już nie (autogenerowany klucz). Oczywiście można by korzystać z jednego i po prostu nie dawać ID w Jsonie, ale wtedy nie można wymusić podawania go przy aktualizacji + trzeba by usunąć @JsonPropertyOrder. Nie jest to jakaś poważna sprawa rzecz jasna, ale dawałoby to większe zabezpieczenie (unikniemy sytuacji, gdy ktoś nie poda ID przy aktualizacji a wtedy trzeba by sprawdzać to i obsługiwać wyjątek).

Reply:
Myślałem nad tym i ew. po prostu można przy update w parametrze podawać id i to by rozwiązało sprawę i po mapowaniu dodać id z paramaterów

Reply:
Jako @PathParam? No można